rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1198,3 +1326,140 @@ def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def update_curve_official_pools_cache(self, write_cursor: DBCursor) -> None:
+        """Query pools from curve registry and save pairs lp_token address - pool address
+        May raise:
+        - RemoteError if failed to query etherscan or node
+        - NotERC20Conformant if failed to query info while calling get_or_create_ethereum_token
+        """
+        get_registry_result = CURVE_ADDRESS_PROVIDER.call(
+            ethereum=self,
+            method_name='get_registry',
+        )
+        registry_address = to_checksum_address(get_registry_result)
+        registry_contract = EthereumContract(
+            address=registry_address,
+            abi=CURVE_REGISTRY_ABI,
+            deployed_block=0,
+        )
+        registry_pool_count = registry_contract.call(
+            ethereum=self,
+            method_name='pool_count',
+        )
+        registry_pools_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='pool_list',
+            all_arguments=list(range(registry_pool_count)),
+            decode=False,  # don't decode since decoding unchecksums address
+        )
+        pool_addresses = []
+        for pool_addr_encoded in registry_pools_result:
+            decoded_pool_addr = hex_or_bytes_to_address(pool_addr_encoded)  # already checksumed
+            pool_addresses.append(decoded_pool_addr)
+
+        registry_lp_tokens_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='get_lp_token',
+            all_arguments=pool_addresses,
+            decode=False,  # don't decode since decoding unchecksums address
+        )
+        registry_coins_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='get_coins',
+            all_arguments=pool_addresses,
+            decode=True,
+        )
+        registry_underlying_coins_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='get_underlying_coins',
+            all_arguments=pool_addresses,
+            decode=True,
+        )
+        pools_mapping: POOLS_MAPPING_TYPE = {}
+        results_zip = zip(pool_addresses, registry_lp_tokens_result, registry_coins_result, registry_underlying_coins_result)  # noqa: E501
+        for pool_addr, lp_token_encoded, decoded_pool_coins, decoded_pool_underlying_coins in results_zip:  # noqa: E501
+            decoded_lp_token = hex_or_bytes_to_address(lp_token_encoded)
+            pool_coins = []
+            pool_underlying_coins = []
+            for pool_coin in decoded_pool_coins:
+                checksumed_coin_addr = to_checksum_address(pool_coin)
```

Comment:
It won't work here since:
1. Coins are tricky encoded (so it is not simply array of 32-bytes values)
2. From this contract they are returned in a lower-case format anyway, so doing `hex_or_bytes_to_address` instead of decoding won't make much sense.